File: packages/api/handlers/typescript/src/utils/exceptions.ts
/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

export class RetryableError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "RetryableError";
  }
}

export class RateLimitError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "RateLimitError";
  }
}

export class ModelResponseError extends RetryableError {
  constructor(message: string) {
    super(message);
    this.name = "ModelResponseError";
  }
}

export class BadRequestError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "BadRequestError";
  }
}

File: packages/smart-product-onboarding/core-utils/amzn_smart_product_onboarding_core_utils/exceptions.py
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: MIT-0

class RetryableError(Exception):
    pass


class RateLimitError(Exception):
    pass


class ModelResponseError(RetryableError):
    pass


class ModelResponseWarning(UserWarning):
    pass


class MaxTokensExceeded(Exception):
    pass
